src/core/PWHistory.h

```diff
@@ -101,6 +101,8 @@ StringX GetPreviousPassword(const StringX &pwh_str);
 
 StringX MakePWHistoryHeader(bool status, size_t pwh_max, size_t pwh_num);
 
+StringX PWHistoryToStringX(PWHistList &pwhistlist, bool keep_history, size_t pwh_max);
```

Comment:
define as a cast operator?

Reply:
Any more feedback on this?  I realized (a bit late) that since I changed core routines, the Windows MFC code may also need an update.  Then again, Windows didn't have the bug in the first place.  I don't have a Windows build environment set-up at the moment to check it.  (I'm hoping someone else can cover that part. ;-)  In any case, I'm working on making PWHistList a full fledged class, the wx side is almost done and looks a bit cleaner, but it will certainly need Windows changes.  If you are interested in that, I think I can submit it as a new PR that would include this one, or as a separate follow-on after this is merged.

Reply:
Since this PR already touches core, and therefore affects the Windows build, I'd like to do the merge once, so please add the changes here.

Reply:
If you want a look at the version with PWHistList as a class, you can see it [here](https://github.com/nobugshere/pwsafe/tree/pwHistClass).  The wx side is done except for coretest.  If you decide you want it, I'll either post a new PR or merge it with this one.  If you don't want to deal with the class changes, I'll add a unit test to this PR.